magma/bit.py

```diff
@@ -135,14 +135,21 @@ def __xor__(self, other):
         return self.declare_binary_op("xor")()(self, other)
 
     def ite(self, t_branch, f_branch):
+        if isinstance(t_branch, list) and isinstance(f_branch, list):
+            if not len(t_branch) == len(f_branch):
+                raise TypeError("Bit.ite of two lists expects the same length")
+            return [self.ite(x, y) for x, y in zip(t_branch, f_branch)]
         t_type = get_type(t_branch)
         f_type = get_type(f_branch)
 
-        # Implicit int conversion
-        if t_type is int and f_type is not int:
+        # Implicit int/bv conversion
+        if (issubclass(t_type, (int, ht.BitVector)) and
+                not issubclass(f_type, (int, ht.BitVector))):
```

Comment:
nit: indent

Reply:
What's the alternative here? If we line it up with the paren, it is 4 tabs in, so the block below will need to be indented, is that preferred?

Reply:
hm thats right, switching between looking at C++ and python gets tough 😩. Fine as is.